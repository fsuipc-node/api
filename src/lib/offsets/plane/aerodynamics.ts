import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const APU: OffsetList = {
  incidenceAlpha: new Offset({
    value: 0x2ED0,
    name: 'incidenceAlpha',
    description: 'incidence alpha (body AoA) - radians',
    category: OffsetCategory.AERODYNAMICS,
    type: Type.Double,
    permission: 'r',
  }),
};
