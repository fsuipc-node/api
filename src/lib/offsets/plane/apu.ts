import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const APU: OffsetList = {
  APUGeneratorSwitch: new Offset({
    value: 0xB51,
    name: 'APUGeneratorSwitch',
    category: OffsetCategory.APU,
    description: 'APU generator switch',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  APUGeneratorActive: new Offset({
    value: 0xB52,
    name: 'APUGeneratorActive',
    category: OffsetCategory.APU,
    description: 'APU generator active',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  APUOnFire: new Offset({
    value: 0xB53,
    name: 'APUOnFire',
    category: OffsetCategory.APU,
    description: 'APU generator active',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  APUMaxRPM: new Offset({
    value: 0xB54,
    name: 'APUMaxRPM',
    category: OffsetCategory.APU,
    description: 'APU generator max RPM percent',
    convert: '+({VAL}).toFixed(2)',
    type: Type.Single,
    permission: 'r',
  }),
  APUStarter: new Offset({
    value: 0xB58,
    name: 'APUStarter',
    category: OffsetCategory.APU,
    description: 'APU starter',
    convert: '{VAL} ? 1 : 0',
    type: Type.Single,
    permission: 'rw',
  }),
  APUVoltage: new Offset({
    value: 0xB5C,
    name: 'APUVoltage',
    category: OffsetCategory.APU,
    description: 'APU generator voltage',
    convert: '+({VAL}).toFixed(2)',
    type: Type.Single,
    permission: 'rw',
  }),
};
