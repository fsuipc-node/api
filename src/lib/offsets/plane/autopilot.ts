import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const autopilot: OffsetList = {
  autoPilotAvailable: new Offset({
    value: 0x764,
    name: 'autoPilotAvailable',
    category: OffsetCategory.AUTOPILOT,
    description: 'autopilot available',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  flyByWireELACSwitch: new Offset({
    value: 0x7B6,
    name: 'flyByWireELACSwitch',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire ELAC switch',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  flyByWireELACCompFailFlag: new Offset({
    value: 0x7B7,
    name: 'flyByWireELACCompFailFlag',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire ELAC comp fail flag (read only)',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  flyByWireFACSwitch: new Offset({
    value: 0x7B8,
    name: 'flyByWireFACSwitch',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire FAC switch',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  flyByWireFACCompFailFlag: new Offset({
    value: 0x7B9,
    name: 'flyByWireFACCompFailFlag',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire FAC comp fail flag (read only)',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  flyByWireSECSwitch: new Offset({
    value: 0x7BA,
    name: 'flyByWireSECSwitch',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire SEC switch',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  flyByWireSECCompFailFlag: new Offset({
    value: 0x7BB,
    name: 'flyByWireSECCompFailFlag',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire SEC comp fail flag (read only)',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  apMasterSwitch: new Offset({
    value: 0x7BC,
    name: 'apMasterSwitch',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP master switch',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apWingLevel: new Offset({
    value: 0x7C0,
    name: 'apWingLevel',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP wing level',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apNav1Hold: new Offset({
    value: 0x7C4,
    name: 'apNav1Hold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP NAV1 hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apHeadingHold: new Offset({
    value: 0x7C8,
    name: 'apHeadingHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP heading hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apHeadingValue: new Offset({
    value: 0x7CC,
    name: 'apHeadingValue',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP heading value - degrees',
    convert: 'Math.round(({VAL} * 360) / 65536)',
    type: Type.UInt16,
    permission: 'rw',
  }),
  apAltitudeHold: new Offset({
    value: 0x7D0,
    name: 'apAltitudeHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP altitude hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apAltitudeValue: new Offset({
    value: 0x7D4,
    name: 'apAltitudeValue',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP altitude value - ft',
    convert: '+({VAL} * 3.28084 / 65536).toFixed(2)',
    type: Type.Int32,
    permission: 'rw',
  }),
  apAttitudeHold: new Offset({
    value: 0x7D8,
    name: 'apAttitudeHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP attitude hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apAsHold: new Offset({
    value: 0x7DC,
    name: 'apAsHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP airspeed hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apAsValue: new Offset({
    value: 0x7E2,
    name: 'apAsValue',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP airspeed value - kt',
    type: Type.UInt16,
    permission: 'rw',
  }),
  apMachHold: new Offset({
    value: 0x7E4,
    name: 'apMachHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP mach hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apMachValue: new Offset({
    value: 0x7E8,
    name: 'apMachValue',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP mach value - mach',
    convert: '{VAL} / 65536',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apVsHold: new Offset({
    value: 0x7EC,
    name: 'apVsHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP vertical speed hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apVsValue: new Offset({
    value: 0x7F2,
    name: 'apVsValue',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP vertical speed value - ft/min',
    type: Type.Int16,
    permission: 'rw',
  }),
  apRPMN1Hold: new Offset({
    value: 0x7F4,
    name: 'apRPMN1Hold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP RPM/N1 hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apRPMN1Value: new Offset({
    value: 0x7FA,
    name: 'apRPMN1Value',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP RPM/N1 value - percent',
    convert: 'Math.round({VAL} * 100 / 16384)',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apGlideSlopeHold: new Offset({
    value: 0x7FC,
    name: 'apGlideSlopeHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP GlideSlope hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apApproachHold: new Offset({
    value: 0x800,
    name: 'apApproachHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP Approach hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apBackCourseHold: new Offset({
    value: 0x804,
    name: 'apBackCourseHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP back course hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apYawDamperHold: new Offset({
    value: 0x808,
    name: 'apYawDamperHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP yaw damper hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apTOGAAutoThrottle: new Offset({
    value: 0x80C,
    name: 'apTOGAAutoThrottle',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP TO/GA throttle hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apAutoThrottleArm: new Offset({
    value: 0x810,
    name: 'apAutoThrottleArm',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP autothrottle arm',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apMaxBank: new Offset({
    value: 0x2E04,
    name: 'apMaxBank',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP max bank - degrees',
    type: Type.UInt32,
    permission: 'r',
  }),
  apMaxSpeed: new Offset({
    value: 0x2F40,
    name: 'apMaxSpeed',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP max speed',
    type: Type.Double,
    permission: 'r',
  }),
  apCruiseSpeed: new Offset({
    value: 0x2F48,
    name: 'apCruiseSpeed',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP cruise speed',
    type: Type.Double,
    permission: 'r',
  }),
};
