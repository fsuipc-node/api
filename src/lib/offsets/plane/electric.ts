import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const electric: OffsetList = {
  masterBatterySwitch: new Offset({
    value: 0x281C,
    name: 'masterBatterySwitch',
    category: OffsetCategory.ELECTRIC,
    description: 'battery master switch',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  totalLoadAmp: new Offset({
    value: 0x2824,
    name: 'totalLoadAmp',
    category: OffsetCategory.ELECTRIC,
    description: 'total load amps',
    type: Type.Double,
    permission: 'r',
  }),
  batteryLoad: new Offset({
    value: 0x282C,
    name: 'batteryLoad',
    category: OffsetCategory.ELECTRIC,
    description: 'battery load',
    type: Type.Double,
    permission: 'r',
  }),
  batteryVoltage: new Offset({
    value: 0x2834,
    name: 'batteryVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'battery voltage',
    type: Type.Double,
    permission: 'r',
  }),
  mainBusVoltage: new Offset({
    value: 0x2840,
    name: 'mainBusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'main bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  mainBusAmps: new Offset({
    value: 0x2848,
    name: 'mainBusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'main bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  avionicsBusVoltage: new Offset({
    value: 0x2850,
    name: 'avionicsBusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'avionics bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  avionicsBusAmps: new Offset({
    value: 0x2858,
    name: 'avionicsBusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'avionics bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  hotBatteryBusVoltage: new Offset({
    value: 0x2860,
    name: 'hotBatteryBusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'hot battery bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  hotBatteryBusAmps: new Offset({
    value: 0x2868,
    name: 'hotBatteryBusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'hot battery bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  batteryBusVoltage: new Offset({
    value: 0x2870,
    name: 'batteryBusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'battery bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  batteryBusAmps: new Offset({
    value: 0x2878,
    name: 'batteryBusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'battery bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator1BusVoltage: new Offset({
    value: 0x2880,
    name: 'generatorAlternator1BusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 1 bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator1BusAmps: new Offset({
    value: 0x2888,
    name: 'generatorAlternator1BusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 1 bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator2BusVoltage: new Offset({
    value: 0x2890,
    name: 'generatorAlternator2BusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 2 bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator2BusAmps: new Offset({
    value: 0x2898,
    name: 'generatorAlternator2BusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 2 bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator3BusVoltage: new Offset({
    value: 0x28A0,
    name: 'generatorAlternator3BusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 3 bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator3BusAmps: new Offset({
    value: 0x28A8,
    name: 'generatorAlternator3BusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 3 bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator4BusVoltage: new Offset({
    value: 0x28B0,
    name: 'generatorAlternator4BusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 4 bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator4BusAmps: new Offset({
    value: 0x28B8,
    name: 'generatorAlternator4BusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 4 bus amps',
    type: Type.Double,
    permission: 'r',
  }),
};
