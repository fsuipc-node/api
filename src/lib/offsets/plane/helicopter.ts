import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const helicopter: OffsetList = {
  rotorbreakActive: new Offset({
    value: 0x81E,
    name: 'rotorbreakActive',
    category: OffsetCategory.HELICOPTER,
    description: 'rotor brake active - Robinson only',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  rotorClutchActive: new Offset({
    value: 0x81F,
    name: 'rotorClutchActive',
    category: OffsetCategory.HELICOPTER,
    description: 'rotor clutch active - Robinson only',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  rotorChipWarning: new Offset({
    value: 0x820,
    name: 'rotorChipWarning',
    category: OffsetCategory.HELICOPTER,
    description: 'rotor chip detected - Rbinson only',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  rotorGovWarning: new Offset({
    value: 0x821,
    name: 'rotorGovWarning',
    category: OffsetCategory.HELICOPTER,
    description: 'rotor gov active - Rbinson only',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  rotorBreakApplication: new Offset({
    value: 0x822,
    name: 'rotorBreakApplication',
    category: OffsetCategory.HELICOPTER,
    description: 'rotor brake application - percent - Robinson only',
    convert: 'Math.round({VAL} / 16384 * 100)',
    type: Type.UInt16,
    permission: 'rw',
  }),
  rotorLateralTrim: new Offset({
    value: 0x824,
    name: 'rotorLateralTrim',
    category: OffsetCategory.HELICOPTER,
    description: 'rotor lateral trim - percent - Robinson only',
    convert: 'Math.round({VAL} / 16384 * 100)',
    type: Type.UInt16,
    permission: 'rw',
  }),
  governorSwitch: new Offset({
    value: 0x826,
    name: 'governorSwitch',
    category: OffsetCategory.HELICOPTER,
    description: 'governor switch activated - Robinson only',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  rotorTransmissionTemp: new Offset({
    value: 0x828,
    name: 'rotorTransmissionTemp',
    category: OffsetCategory.HELICOPTER,
    description: 'rotor transmission temp - degrees Rankine - read only',
    type: Type.Double,
    permission: 'r',
  }),
  helicopterEngine1RPM: new Offset({
    value: 0x896,
    name: 'helicopterEngine1RPM',
    category: OffsetCategory.HELICOPTER,
    description: 'engine 1 RPM %',
    type: Type.UInt16,
    permission: 'rw',
  }),
  helicopterEngine1RPMUnscaled: new Offset({
    value: 0x898,
    name: 'helicopterEngine1RPMUnscaled',
    category: OffsetCategory.HELICOPTER,
    description: 'engine 1 RPM unscalled - multiply by engineRPMScaler to obtain derivated RPM',
    type: Type.UInt16,
    permission: 'rw',
  }),
  helicopterEngine1ElectricalLoad: new Offset({
    value: 0x8FC,
    name: 'helicopterEngine1ElectricalLoad',
    category: OffsetCategory.HELICOPTER,
    description: 'engine1 electrical load - trust only for helo - percent',
    convert: '{VAL} / 16384 * 100',
    type: Type.UInt32,
    permission: 'r',
  }),
  helicopterEngine1TransmOilPres: new Offset({
    value: 0x900,
    name: 'helicopterEngine1TransmOilPres',
    category: OffsetCategory.HELICOPTER,
    description: 'engine1 transmission oil pressure - trust only for helo - PSI',
    convert: '{VAL} / 16384',
    type: Type.UInt32,
    permission: 'r',
  }),
  helicopterEngine1TransmOilTemp: new Offset({
    value: 0x904,
    name: 'helicopterEngine1TransmOilTemp',
    category: OffsetCategory.HELICOPTER,
    description: 'engine1 transmission oil temperature - trust only for helo - C',
    convert: '{VAL} / 16384',
    type: Type.UInt32,
    permission: 'r',
  }),
  helicopterEngine1RotorRPM: new Offset({
    value: 0x908,
    name: 'helicopterEngine1RotorRPM',
    category: OffsetCategory.HELICOPTER,
    description: 'engine1 rotor RPM - percent',
    convert: '{VAL} / 16384 * 100',
    type: Type.UInt32,
    permission: 'r',
  }),
  helicopterPitchTrim: new Offset({
    value: 0xBBE,
    name: 'helicopterPitchTrim',
    description: 'helicopter pitch trim',
    category: OffsetCategory.HELICOPTER,
    convert: 'Math.round({VAL} / 16383 * 100)',
    type: Type.Int16,
    permission: 'rw',
  }),
  helicopterBankTrim: new Offset({
    value: 0xC06,
    name: 'helicopterBankTrim',
    description: 'helicopter bank trim - only when apply hello trim to both',
    category: OffsetCategory.HELICOPTER,
    convert: 'Math.round({VAL} / 16383 * 100)',
    type: Type.Int16,
    permission: 'rw',
  }),
};
