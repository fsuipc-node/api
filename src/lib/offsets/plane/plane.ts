import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const plane: OffsetList = {
  vs0: new Offset({
    value: 0x538,
    name: 'vs0',
    category: OffsetCategory.PLANE,
    description: 'Stall speed full flaps - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  vs1: new Offset({
    value: 0x540,
    name: 'vs1',
    category: OffsetCategory.PLANE,
    description: 'Stall speed clean - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  vc: new Offset({
    value: 0x548,
    name: 'vc',
    category: OffsetCategory.PLANE,
    description: 'Cruise speed - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  vmd: new Offset({
    value: 0x550,
    name: 'vmd',
    category: OffsetCategory.PLANE,
    description: 'Minimum drag speed - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  smokeSystemControl: new Offset({
    value: 0x5D8,
    name: 'smokeSystemControl',
    category: OffsetCategory.PLANE,
    convert: '!!{VAL}',
    description: 'smoke system control on/off',
    type: Type.UInt16,
    permission: 'rw',
  }),
  retractableGear: new Offset({
    value: 0x60C,
    name: 'retractableGear',
    category: OffsetCategory.PLANE,
    description: 'retractable gear (read only)',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  retractableLeftFloatExtension: new Offset({
    value: 0x614,
    name: 'retractableLeftFloatExtension',
    category: OffsetCategory.PLANE,
    description: 'retractable left float extension - percent',
    convert: 'Math.floor({VAL} / 16384 * 100)',
    type: Type.UInt16,
    permission: 'r',
  }),
  retractableRightFloatExtension: new Offset({
    value: 0x616,
    name: 'retractableRightFloatExtension',
    category: OffsetCategory.PLANE,
    description: 'retractable right float extension - percent',
    convert: 'Math.floor({VAL} / 16384 * 100)',
    type: Type.UInt16,
    permission: 'r',
  }),
  hasFlaps: new Offset({
    value: 0x778,
    name: 'hasFlaps',
    category: OffsetCategory.PLANE,
    description: 'has flaps',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasStallHorn: new Offset({
    value: 0x77C,
    name: 'hasStallHorn',
    category: OffsetCategory.PLANE,
    description: 'has stall horn',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasMixtureEngine: new Offset({
    value: 0x780,
    name: 'hasMixtureEngine',
    category: OffsetCategory.PLANE,
    description: 'has mixture engine',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasCarbHeat: new Offset({
    value: 0x784,
    name: 'hasCarbHeat',
    category: OffsetCategory.PLANE,
    description: 'has carburator heater',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasSpoilers: new Offset({
    value: 0x78C,
    name: 'hasSpoilers',
    category: OffsetCategory.PLANE,
    description: 'has spoilers',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  isTailDragger: new Offset({
    value: 0x790,
    name: 'isTailDragger',
    category: OffsetCategory.PLANE,
    description: 'has spoilers',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasStrobes: new Offset({
    value: 0x794,
    name: 'hasStrobes',
    category: OffsetCategory.PLANE,
    description: 'has strobes',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasToeBrakes: new Offset({
    value: 0x79C,
    name: 'hasToeBrakes',
    category: OffsetCategory.PLANE,
    description: 'has toe brakes',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  maxMach: new Offset({
    value: 0xB0C,
    name: 'maxMach',
    category: OffsetCategory.PLANE,
    description: 'mach maximum operating - not FS2004',
    convert: '+({VAL} / 20438).toFixed(2)',
    type: Type.Int32,
    permission: 'r',
  }),
  fuelSelectorsCount: new Offset({
    value: 0x123E,
    name: 'fuelSelectorsCount',
    category: OffsetCategory.PLANE,
    description: 'number of fuel selectors for the aircraft',
    type: Type.Byte,
    permission: 'r',
  }),
  aircraftEmptyWeight: new Offset({
    value: 0x1330,
    name: 'aircraftEmptyWeight',
    category: OffsetCategory.PLANE,
    description: 'aicraft empty weight',
    convert: '+({VAL} / 256).toFixed(2)',
    type: Type.Int32,
    permission: 'r',
  }),
  aircraftMaxGrossWeight: new Offset({
    value: 0x1334,
    name: 'aircraftMaxGrossWeight',
    category: OffsetCategory.PLANE,
    description: 'aicraft max gross weight',
    convert: '+({VAL} / 256).toFixed(2)',
    type: Type.Int32,
    permission: 'r',
  }),
};
