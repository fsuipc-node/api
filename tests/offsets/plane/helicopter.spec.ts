import { helicopter as offsets } from '@offsets/plane/helicopter';

describe('offset - plane/helicopter', () => {
  const offsetsTestCases = [
    { name: 'rotorBreakApplication', value: 13107, expectedResult: 80 },
    { name: 'rotorLateralTrim', value: 13107, expectedResult: 80 },
    { name: 'governorSwitch', value: 0, expectedResult: false },
    { name: 'governorSwitch', value: 1, expectedResult: true },
    { name: 'helicopterEngine1ElectricalLoad', value: 8192, expectedResult: 50 },
    { name: 'helicopterEngine1TransmOilPres', value: 3276800, expectedResult: 200 },
    { name: 'helicopterEngine1TransmOilTemp', value: 3276800, expectedResult: 200 },
    { name: 'helicopterEngine1RotorRPM', value: 8192, expectedResult: 50 },
  ];

  describe('offsets list', () => {
    it('should have required properties', () => {
      expect(offsets).toMatchSnapshot();
    });
  });

  offsetsTestCases.forEach(testedOffset => {
    describe(testedOffset.name, () => {
      it('should convert data properly', () => {
        const convertExpression = offsets[testedOffset.name].convert.replace(new RegExp(/{VAL}/g), testedOffset.value.toString());

        // tslint:disable-next-line:no-eval
        expect(eval(convertExpression)).toEqual(testedOffset.expectedResult);
      });
    });
  });
});
