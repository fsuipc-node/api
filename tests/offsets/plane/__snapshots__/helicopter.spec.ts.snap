// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offset - plane/helicopter offsets list should have required properties 1`] = `
Object {
  "governorSwitch": Offset {
    "category": "helicopter",
    "convert": "!!{VAL}",
    "description": "governor switch activated - Robinson only",
    "length": undefined,
    "mapping": undefined,
    "name": "governorSwitch",
    "permission": "rw",
    "type": 0,
    "value": 2086,
  },
  "helicopterEngine1ElectricalLoad": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384 * 100",
    "description": "engine1 electrical load - trust only for helo - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "helicopterEngine1ElectricalLoad",
    "permission": "r",
    "type": 6,
    "value": 2300,
  },
  "helicopterEngine1RPM": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 RPM %",
    "length": undefined,
    "mapping": undefined,
    "name": "helicopterEngine1RPM",
    "permission": "rw",
    "type": 5,
    "value": 2198,
  },
  "helicopterEngine1RPMUnscaled": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 RPM unscalled - multiply by engineRPMScaler to obtain derivated RPM",
    "length": undefined,
    "mapping": undefined,
    "name": "helicopterEngine1RPMUnscaled",
    "permission": "rw",
    "type": 5,
    "value": 2200,
  },
  "helicopterEngine1RotorRPM": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384 * 100",
    "description": "engine1 rotor RPM - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "helicopterEngine1RotorRPM",
    "permission": "r",
    "type": 6,
    "value": 2312,
  },
  "helicopterEngine1TransmOilPres": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384",
    "description": "engine1 transmission oil pressure - trust only for helo - PSI",
    "length": undefined,
    "mapping": undefined,
    "name": "helicopterEngine1TransmOilPres",
    "permission": "r",
    "type": 6,
    "value": 2304,
  },
  "helicopterEngine1TransmOilTemp": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384",
    "description": "engine1 transmission oil temperature - trust only for helo - C",
    "length": undefined,
    "mapping": undefined,
    "name": "helicopterEngine1TransmOilTemp",
    "permission": "r",
    "type": 6,
    "value": 2308,
  },
  "rotorBreakApplication": Offset {
    "category": "helicopter",
    "convert": "Math.round({VAL} / 16384 * 100)",
    "description": "rotor brake application - percent - Robinson only",
    "length": undefined,
    "mapping": undefined,
    "name": "rotorBreakApplication",
    "permission": "rw",
    "type": 5,
    "value": 2082,
  },
  "rotorChipWarning": Offset {
    "category": "helicopter",
    "convert": "!!{VAL}",
    "description": "rotor chip detected - Rbinson only",
    "length": undefined,
    "mapping": undefined,
    "name": "rotorChipWarning",
    "permission": "rw",
    "type": 0,
    "value": 2080,
  },
  "rotorClutchActive": Offset {
    "category": "helicopter",
    "convert": "!!{VAL}",
    "description": "rotor clutch active - Robinson only",
    "length": undefined,
    "mapping": undefined,
    "name": "rotorClutchActive",
    "permission": "rw",
    "type": 0,
    "value": 2079,
  },
  "rotorGovWarning": Offset {
    "category": "helicopter",
    "convert": "!!{VAL}",
    "description": "rotor gov active - Rbinson only",
    "length": undefined,
    "mapping": undefined,
    "name": "rotorGovWarning",
    "permission": "rw",
    "type": 0,
    "value": 2081,
  },
  "rotorLateralTrim": Offset {
    "category": "helicopter",
    "convert": "Math.round({VAL} / 16384 * 100)",
    "description": "rotor lateral trim - percent - Robinson only",
    "length": undefined,
    "mapping": undefined,
    "name": "rotorLateralTrim",
    "permission": "rw",
    "type": 5,
    "value": 2084,
  },
  "rotorTransmissionTemp": Offset {
    "category": "helicopter",
    "convert": undefined,
    "description": "rotor transmission temp - degrees Rankine - read only",
    "length": undefined,
    "mapping": undefined,
    "name": "rotorTransmissionTemp",
    "permission": "r",
    "type": 8,
    "value": 2088,
  },
  "rotorbreakActive": Offset {
    "category": "helicopter",
    "convert": "!!{VAL}",
    "description": "rotor brake active - Robinson only",
    "length": undefined,
    "mapping": undefined,
    "name": "rotorbreakActive",
    "permission": "rw",
    "type": 0,
    "value": 2078,
  },
}
`;
